### DISCUSSION TITLE
How to set term items on uploaded files in SharePoint via Graph API

### DISCUSSION THREAD

**Discussion by M-Patrone** [microsoftgraph/msgraph-sdk-dotnet#2600] [Q&A]
Hello everyone

I hope you can help me with a specific issue I am encountering.

I am currently working on uploading files to SharePoint via the Microsoft Graph API and need to update term items on these files once they are uploaded. While I have successfully figured out how to upload a file to SharePoint using the Graph API with application permission flow, I am running into problems when trying to set term items.

From what I've gathered, it seems that setting term items might not be possible with application permissions, though I found a discussion that suggests it could be managed in some way ([Create term group in term store using Microsoft Graph API](https://stackoverflow.com/questions/72966749/create-term-group-in-term-store-using-microsoft-graph-api)). Has anyone here had experience with this?

My main challenge is understanding how to set multiple term items on an uploaded document in SharePoint. I am unsure whether I am using the correct API call for this purpose. Specifically, I've looked at this documentation: [Update term in term store using Microsoft Graph API](https://learn.microsoft.com/en-us/graph/api/termstore-term-update?view=graph-rest-1.0&tabs=http), but I can't find any parameters indicating how to specify the document the term items should be applied to.

For context, I use the `LargeFileUploadTask<DriveItem>` to upload documents. I've seen several forum posts suggesting that it might not be possible to set term items at all. Can anyone confirm whether this is true or provide a solution? Is the API call mentioned above the correct one to use for setting term items on a document?

Any guidance or examples from those who have tackled similar issues would be greatly appreciated.

Thank you!

**Comment by andrueastman** [comment]
Thanks for raising this @M-Patrone 

Any chance you can post the same question at the link below? As this question is mainly API related, and this repo is mainly for SDK related question, we may not be best placed to answer this.

https://aka.ms/askGraph
